### PR TITLE
Improve the way sysbox-mgr configures the /etc/subuid file.

### PR DESCRIPTION
**Pull request by ctalledo**
This change improves the way sysbox-mgr configures the /etc/subuid
and /etc/subgid files. Prior to this change, it was possible
for sysbox-mgr to inadvertendly add rendundand entries for
user "sysbox" into these files. After this change this is
no longer possible.

Signed-off-by: Cesar Talledo <ctalledo@nestybox.com>

### CODE REVIEW

**Review by rodnymolina — COMMENTED** [review]
Question. With this code (and also with the previous one), it looks like we are changing subuid and subgid ranges regardless of the mode in which we are operating (shiftfs or userns-remap).

This wouldn't be a problem in shiftfs setups as Sysbox is allocating the ids dynamically. But in userns-remap mode, looks like we shouldn't touch these files whenever Sysbox is restarted, as that may prevent subsequent instantiations of Sysbox from being able to access external resources/volumes for which an uid-mapping was previously defined.

In short, looks like we need some uid/gid stability in userns-remap mode, isn't it?

**Comment by ctalledo** [comment]
> Question. With this code (and also with the previous one), it looks like we are changing subuid and subgid ranges regardless of the mode in which we are operating (shiftfs or userns-remap).
> 
> This wouldn't be a problem in shiftfs setups as Sysbox is allocating the ids dynamically. But in userns-remap mode, looks like we shouldn't touch these files whenever Sysbox is restarted, as that may prevent subsequent instantiations of Sysbox from being able to access external resources/volumes for which an uid-mapping was previously defined.
> 
> In short, looks like we need some uid/gid stability in userns-remap mode, isn't it?

That's a good observation. In the prior version of the code, we did have stability for the subid ranges for user `sysbox` because the code was first checking if a valid entry for that user was found and leaving untouched if so. The new version does not do this, so that's not as good. I'll fix it. We don't need to differentiate between userns-remap or shiftfs, we should try to keep stability of the subuid ranges in either (although it matters more for userns-remap as you indicated).

Let me fix and repost.

**Comment by ctalledo** [comment]
> Let me fix and repost.

I've fixed and reposted the diffs. I also added a new unit test to verify that Sysbox keeps existing configs for user "sysbox" in `/etc/sub[ug]id`. Thanks!

**Review by rodnymolina — APPROVED** [review]
LG!
